core/src/main/java/io/questdb/cutlass/text/TextImportTask.java

```diff
@@ -0,0 +1,992 @@
+            } catch (Exception ignore) {
+                errors++;
+                logError(offset, fieldIndex, dbcs);
+                switch (atomicity) {
+                    case Atomicity.SKIP_ALL:
+                        tableWriterRef.rollback();
+                        throw TextException.$("bad syntax [line offset=").put(offset).put(",column=").put(fieldIndex).put(']');
+                    case Atomicity.SKIP_ROW:
+                        w.cancel();
+                        return true;
+                    default: // SKIP column
+                        break;
+                }
+            }
+            return false;
+        }
+
+        private void onFieldsPartitioned(long line, ObjList<DirectByteCharSequence> values, int valuesLength) {
+            assert tableWriterRef != null;
+            DirectByteCharSequence dbcs = values.getQuick(timestampIndex);
+            final TableWriter.Row w = getRow(dbcs, offset);
+            if (w == null) {
+                return;
+            }
+            for (int i = 0; i < valuesLength; i++) {
+                dbcs = values.getQuick(i);
+                if (i == timestampIndex || dbcs.length() == 0) {
+                    continue;
+                }
+                if (onField(offset, dbcs, w, i, utf8Sink)) return;
+            }
+            w.append();
+        }
+
+        private TableWriter.Row getRow(DirectByteCharSequence dbcs, long offset) {
+            try {
+                return tableWriterRef.newRow(timestampAdapter.getTimestamp(dbcs));
+            } catch (Exception e) {
+                if (atomicity == Atomicity.SKIP_ALL) {
+                    throw TextException.$("Can't parse timestamp on line starting at offset=").put(offset).put(". ").put(e.getMessage());
+                } else {
+                    logError(offset, timestampIndex, dbcs);
+                    return null;
+                }
+            }
+        }
+
+        private long openIndexChunks(FilesFacade ff, Path partitionPath, DirectLongList mergeIndexes, int partitionLen) {
+            long mergedIndexSize = 0;
+            long chunk = ff.findFirst(partitionPath);
+            if (chunk > 0) {
+                try {
+                    do {
+                        // chunk loop
+                        long chunkName = ff.findName(chunk);
+                        long chunkType = ff.findType(chunk);
+                        if (chunkType == Files.DT_FILE) {
+                            partitionPath.trimTo(partitionLen);
+                            partitionPath.concat(chunkName).$();
+
+                            long fd = TableUtils.openRO(ff, partitionPath, LOG);
+                            long size = 0;
+                            long address = -1;
+
+                            try {
+                                size = ff.length(fd);
+                                address = TableUtils.mapRO(ff, fd, size, MemoryTag.MMAP_PARALLEL_IMPORT);
```

Comment:
can size be < 1 ?

Reply:
Chunk shouldn't be created if there's no data to put into it so no but writes are async via memory class . 
I'll add exception to stop import in such case because it'd mean something messed up files  .